pkg/build/util/generate_test.go

```diff
@@ -352,3 +532,22 @@ func mockBuildConfig(source api.BuildSource, strategy api.BuildStrategy, output
 		},
 	}
 }
+
+type mockImageRepositoryNamespaceGetter struct {
+	namespace string
+	name      string
+}
+
+func (m *mockImageRepositoryNamespaceGetter) GetByNamespace(namespace, name string) (*imageapi.ImageRepository, error) {
+	if m.namespace == namespace && m.name == name {
+		return &imageapi.ImageRepository{
+			ObjectMeta: kapi.ObjectMeta{Name: imageRepoName},
+			Status: imageapi.ImageRepositoryStatus{
+				DockerImageRepository: originalImage,
+			},
+			Tags: map[string]string{tagName: newTag},
+		}, nil
+	} else {
```

Comment:
Don't use else blocks when the previous block returns, and you should probably invert the block so that the exceptional case exits early.  Functions should have the largest amount of code with fewest indentations.